### DISCUSSION TITLE
Can you forward open() calls to socket inodes to a backing socket on another filesystem?

### DISCUSSION THREAD

**Discussion by Qix-** [libfuse/libfuse#681] [Q&A] [answered]
If I have a FUSE FS that is responding to a `getattr` call with a socket node type, is it possible to eschew buffer copies when an application `open`s and interacts with it?

More specifically, if my FS is aware of a backing FS somewhere else in the system that has the _real_ socket file, is there a way to re-direct or otherwise forward an `open()` to that socket file instead of my own, without using symlinks?

Symlinks would otherwise work, but I was hoping to be able to prevent direct lookup and access of backing sockets from an application interacting with the FUSE mountpoint (this is of course not the _only_ function of the FUSE FS daemon, this is just something I'd like to solve elegantly).

**Comment by bsbernd** [comment]
Do you mean something like fuse-passthrough, which is currently in work? I.e. reads/writes are directly handled by the through the underlying file system, instead of fuse?

**Reply by Qix-** [reply]
Correct, yes, I believe that was my goal at the time. I wanted to use FUSE almost as an overlay on top of another mounted filesystem.

**Comment by bsbernd** [comment, marked as answer]
This will be hopefully get merged in the next kernel merge window (6.9)

https://lore.kernel.org/linux-fsdevel/20240206142453.1906268-1-amir73il@gmail.com/
